new-frontiers-14/frontier-station-14
# Loadouts: fallback to free items when character cannot afford mandatory loadout parts.

## About the PR



This PR adds an optional "fallbacks" field to loadoutGroups.  These will be provided to a character, up to their minimum, in the order specified, if the user cannot afford their selected items in any categories.

Also, a character can now buy items that cost their entire bank account (including free items if they have no money).

## Why / Balance


On the current head of master, a completely bankrupt character will spawn nude, with no PDA or jumpsuit, which is not great.

This approach allows more flexibility in the set of items chosen, but relies on developers to maintain a fallback list for loadouts.  As an alternative, we could select the first N items up to minCount instead.

Also, removed redundant backpack definitions from the mercenary loadouts.

## How to test


1. Spawn in-game.
2. Withdraw all your money from an ATM.
3. Run /golobby
4. Pick a Contractor loadout with expensive items in each category you can.  Select multiple communications keys.
5. Spawn in-game as a Contractor.
6. You should have a jumpsuit, shoes, a PDA, a backpack, a survival kit, and the communications keys you selected.

## Media


https://github.com/new-frontiers-14/frontier-station-14/assets/166147148/28a63708-4d30-4842-a70c-0557c973d1db


## Breaking changes


**Changelog**



- fix: Characters with unaffordable loadouts will receive free items to fill mandatory categories (e.g. PDA, jumpsuit).
